ciphers/trifid_cipher.py

```diff
@@ -110,13 +216,35 @@ def encrypt_message(
 def decrypt_message(
     message: str, alphabet: str = "ABCDEFGHIJKLMNOPQRSTUVWXYZ.", period: int = 5
 ) -> str:
+    """
+    decrypt_message
+    ===============
+
+    Decrypts a trifid_cipher encrypted message .
+
+    PARAMETERS
+    ----------
+
+    *   message: The message you want to decrypt .
+    *   alphabet (optional): The characters used for the cipher.
+    *   period (optional): The number of characters used in grouping when it
+        was encrypted.
+
+    >>> decrypt_message('BCDGBQY')
+    'IAMABOY'
+
+    Decrypting with your own alphabet and period
+    >>> decrypt_message('FMJFVOISSUFTFPUFEQQC','FELIXMARDSTBCGHJKNOPQUVWYZ+',5)
+    'AIDETOILECIELTAIDERA'
+    """
     message, alphabet, character_to_number, number_to_character = __prepare(
         message, alphabet
     )
+
     decrypted_numeric = []
     decrypted = ""
 
-    for i in range(0, len(message) + 1, period):
+    for i in range(0, len(message), period):

```

Comment:
Why this change??

Reply:
For situations where the length of the string is a multiple of the period, the loop continues even after it had finished traversing the string, there by creating a ' list index out of range ' error in the __decrypt_part helper function. To reproduce the bug try
```python
decrypt_message('FMJFVOISSUFTFPUFEQQC','FELIXMARDSTBCGHJKNOPQUVWYZ+',5)
# Expected 'AIDETOILECIELTAIDERA'

# The length of the string is 20, a multiple of the period 5.

for i in range(0, (20+1), period)
# This will iterate 5 times [0, 5, 10, 15, 20], taking 5 letters of the string 5 times.
# The 5th time will be an empty string because 5*5 = 25 and we have just 20 letters
# Creating a 'list index out of range' error

for i in range(0, (20) , period)
# Will iterate 4 times [0, 5 10, 15], which is what we want.
# It will also work for strings with lengths that are not multiples of the period




```